yb-voyager/src/dbzm/valueConverter.go

```diff
@@ -0,0 +1,150 @@
+	return row, nil
+}
+
+func (nvc *NoOpValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	return nil
+}
+
+//============================================================================
+
+type DebeziumValueConverter struct {
+	schemaRegistry      *SchemaRegistry
+	valueConverterSuite map[string]tgtdb.ConverterFn
+	converterFnCache    map[string][]tgtdb.ConverterFn
+	snapshotMode        bool
+}
+
+func NewDebeziumValueConverter(exportDir string, tdb tgtdb.TargetDB, snapshotMode bool) (*DebeziumValueConverter, error) {
+	schemaRegistry := NewSchemaRegistry(exportDir)
+	err := schemaRegistry.Init()
+	if err != nil {
+		return nil, fmt.Errorf("initializing schema registry: %w", err)
+	}
+	tdbValueConverterSuite := tdb.GetDebeziumValueConverterSuite(snapshotMode)
+
+	return &DebeziumValueConverter{
+		schemaRegistry:      schemaRegistry,
+		valueConverterSuite: tdbValueConverterSuite,
+		converterFnCache:    map[string][]tgtdb.ConverterFn{},
+		snapshotMode:        snapshotMode,
+	}, nil
+}
+
+func (conv *DebeziumValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	converterFns, err := conv.getConverterFns(tableName, columnNames)
+	if err != nil {
+		return "", fmt.Errorf("fetching converter functions: %w", err)
+	}
+	columnValues := strings.Split(row, "\t")
+	for i, columnValue := range columnValues {
+		if columnValue != "\\N" {
+			transformedValue, err := converterFns[i](columnValue, false)
+			if err != nil {
+				return "", fmt.Errorf("converting value: %w", err)
+			}
+			columnValues[i] = transformedValue
+		} else {
+			columnValues[i] = columnValue
+		}
+	}
+	return strings.Join(columnValues, "\t"), nil
+}
+
+func (conv *DebeziumValueConverter) getConverterFns(tableName string, columnNames []string) ([]tgtdb.ConverterFn, error) {
+	result := conv.converterFnCache[tableName]
+	if result == nil {
+		colTypes, err := conv.schemaRegistry.GetColumnTypes(tableName, columnNames)
+		if err != nil {
+			return nil, fmt.Errorf("fetching column schemas: %w", err)
+		}
+		result = make([]tgtdb.ConverterFn, len(columnNames))
+		for i := range columnNames {
+			colType := colTypes[i]
+			if conv.valueConverterSuite[colType] != nil {
+				result[i] = conv.valueConverterSuite[colType]
+			} else {
+				result[i] = func(v string, formatIfRequired bool) (string, error) { return v, nil }
+			}
+		}
+		conv.converterFnCache[tableName] = result
+	}
+	return result, nil
+}
+
+func (conv *DebeziumValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	err := conv.convertMap(table, ev.Key)
+	if err != nil {
+		return fmt.Errorf("convert event fields: %w", err)
+	}
+	err = conv.convertMap(table, ev.Fields)
+	if err != nil {
+		return fmt.Errorf("convert event keys: %w", err)
+	}
+	return nil
+}
+
+func (conv *DebeziumValueConverter) convertMap(tableName string, m map[string]string) error {
+	for column, value := range m {
+		if value == "" {
+			m[column] = "NULL"
```

Comment:
Nope!! Consider an update event that cleared value of a string column by setting it to `""`. This line will result in writing `"NULL"` in the target table.

Debezium plugin must make the distinction of whether to use empty string or `"NULL"`. Voyager should not handle this case.